electron-cash

```diff
@@ -425,12 +425,6 @@ if __name__ == '__main__':
                 if config.get('websocket_server'):
                     from electroncash import websockets
                     websockets.WebSocketServer(config, d.network).start()
-                if config.get('requests_dir'):
-                    path = os.path.join(config.get('requests_dir'), 'index.html')
-                    if not os.path.exists(path):
-                        print("Requests directory not configured.")
-                        print("You can configure it using https://github.com/spesmilo/electrum-merchant")
-                        sys.exit(1)
```

Comment:
Ok -- I still have to check if other parts of the code will crash if they don't find the requests_dir when they expected it

I do agree checking for `index.html` here is kind of dumb.

So you are 100% sure no check here is needed?

Reply:
Well, as far as I know, electrum-merchant just installs its script there with some other jquery files and stuff like that. On the one hand, enforcing this restricts users to serving their payment requests with electrum-merchant. On the other hand, not enforcing it doesn't break electrum merchant (the user will simply get a broken link when they visit the index_url if electrum-merchant is not installed) but allows users to have a requests_dir and also serve the PR with any URL they want.

Reply:
Gotcha.

The thing is `wallet.py`  also refers to that index.html file:

https://github.com/Electron-Cash/Electron-Cash/blob/1f436b5b21ae6962fa625dfb64d36189d3301a59/lib/wallet.py#L1940

So -- what do we do?

Reply:
Although if you want to continue to support https://electroncash.readthedocs.io/en/latest/old/merchant.html then that doc should be updated to indicate the need to either install electrum-merchant OR serve the PR some other way.

Reply:
It should also explain how to specify BCH when installing electrum-merchant. But imho that static html tool is not great because users can simply spoof their own payments by sending a 'paid' message through the websocket.

Reply:
> Gotcha.
> 
> The thing is `wallet.py` also refers to that index.html file:
> 
> https://github.com/Electron-Cash/Electron-Cash/blob/1f436b5b21ae6962fa625dfb64d36189d3301a59/lib/wallet.py#L1940
> 
> So -- what do we do?

Yeah, because that index_url is included as part of the PR json file. But I don't know that it ever gets read from anywhere after that. Does it just sit there doing nothing? I think so. When the user's webserver gets a request for mydomain.com/requests_dir/index.html?id=FOO, the PR will be accessed by the javascript in electrum-merchant (https://github.com/spesmilo/electrum-merchant/blob/master/electrum-merchant/simple/index.html) without ever needing to access index_url from the JSON file. Pretty sure that field does nothing. You could remove that line as well. (Hope I'm right about that haha)